portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,13 +56,21 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("comment");
-    PreparedQuery results = datastore.prepare(query);
-    ArrayList<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String)entity.getProperty("value"));
+    String maxCommentsString = request.getParameter("max_comments");
+    if (maxCommentsString != null) {
+      Integer maxComments = Integer.parseInt(maxCommentsString);
+      Query query = new Query(COMMENT_TAG);
```

Comment:
minor nit: you are tying your database key and query key to the same name by using the same constant. fine for now, but something to think about in terms of how code might change in the future.